src/app/units/states/edit/directives/unit-staff-editor/unit-staff-editor.component.html

```diff
@@ -0,0 +1,108 @@
+        </mat-card-header>
+        <mat-card-content>
+            <table
+                mat-table [dataSource]="dataSource"
+                matSort
+                class="f-table"
+            >
+                <!-- Name Column-->
+                <ng-container matColumnDef="name">
+                    <th mat-header-cell *matHeaderCellDef mat-sort-header>Name</th>
+                    <td mat-cell *matCellDef="let staff">
+                      <span class="staff-name-container">
+                        <user-icon [size]="30" [user]="staff.user"></user-icon>
+                        <p>{{staff.user.name}}</p>
+                      </span>
+                    </td>
+                </ng-container>
+                
+                <!-- Role Column -->
+        <ng-container matColumnDef="role" sticky>
+            <th mat-header-cell *matHeaderCellDef mat-sort-header>Role</th>
+            <td mat-cell *matCellDef="let staff">
+              
+              <mat-button-toggle-group name="fontStyle" aria-label="Font Style" [(ngModel)]="staff.role" value="{{staff.role}}">
+                <mat-button-toggle value="Tutor"  (click)="changeRole(staff,2)" matTooltip="Tutors deliver unit content">Tutor</mat-button-toggle>
+                <mat-button-toggle value="Convenor"  (click)="changeRole(staff,3)" matTooltip="Convenors manage the unit">Convenor</mat-button-toggle>
+              </mat-button-toggle-group>
+            </td>
+          </ng-container>
+  
+          <!-- Main Convenor Toggle Column-->
+          <ng-container matColumnDef="mainConvenor" sticky>
+            <th mat-header-cell *matHeaderCellDef mat-sort-header>Main Convenor</th>
+            <td mat-cell *matCellDef="let staff">
+              <button
+                ng-show="staff.user.role == 'Convenor'"
+                ng-disabled="staff.id == unit.mainConvenor.id"
+                type="button"
+                class="btn btn-default"
+                [(ng-model)]="unit.mainConvenor.id"
+                (click)="changeMainConvenor(staff)"
+              >
+              <i class="fa fa-{{staff.id == unit.mainConvenor.id ? 'check' : 'times'}}"></i>
+
+            </button>
+            </td>
+          </ng-container>
+  
+    
+          <!-- Action Column-->
+          <ng-container matColumnDef="test">
+            <th mat-header-cell *matHeaderCellDef mat-sort-header>Actions</th>
+            <td style="width: 30px">
+            </td>
+            <td mat-cell *matCellDef="let staff">
+              <button class="delete-button" mat-flat-button (click)="removeStaff(staff)">
+                <i class="fa fa-trash-o"></i>
+              </button>
+            </td>
+          </ng-container>
+                <tr mat-header-row *matHeaderRowDef="columns"></tr>
+                <tr mat-row *matRowDef="let row; columns: columns" class="table-row"></tr>
```

Comment:
Provide a placeholder for the staff autocomplete input field to guide users on what to enter. This improves usability and clarity.